code/datums/status_effects/debuffs/fire_stacks.dm

```diff
@@ -208,6 +210,16 @@
 		return
 
 	victim.adjust_bodytemperature((BODYTEMP_HEATING_MAX + (stacks * 12)) * 0.5 * seconds_per_tick)
+	switch(ticks_on_fire)
+		if(0 to 3)
+			victim.apply_damage(0.5 * stacks, BURN)
+		if(3 to 6)
+			victim.apply_damage(1 * stacks, BURN)
+		if(6 to 9)
+			victim.apply_damage(3 * stacks, BURN)
+		if(10 to INFINITY)
+			victim.apply_damage(5 * stacks, BURN)
+	ticks_on_fire += 1 * seconds_per_tick
```

Comment:
It's not really ticks_on_fire if it scales with seconds_per_tick. Wouldn't a better name be seconds_on_fire?